linera-chain/src/chain.rs

```diff
@@ -345,14 +345,7 @@ where
     /// Removes the incoming messages in the block from the inboxes.
     pub async fn remove_events_from_inboxes(&mut self, block: &Block) -> Result<(), ChainError> {
         let chain_id = self.chain_id();
-        let origins = block
-            .incoming_messages
-            .clone()
-            .into_iter()
-            .map(|message| message.origin)
-            .collect::<Vec<_>>();
-        let inboxes = self.inboxes.try_load_entries_mut(origins).await?;
```

Comment:
This one is interesting! The problem is that the same origin can occur several times.
So you would need to start with something like that:
```
let origins = block
            .incoming_messages
            .clone()
            .into_iter()
            .map(|message| message.origin)
            .collect::<HashSet<_>>();
 ```
Then, this case would really require `try_load_entries_mut` to return an `impl Iterator<K, &mut View>` that we would collect into a map.

Reply:
To be clear, I think this can be done in a next PR.

Reply:
Ok thank you.